addons/website_sale_autocomplete/static/src/js/addressForm.js

```diff
@@ -0,0 +1,88 @@
+/** @odoo-module */
+
+import publicWidget from 'web.public.widget';
+import { DropPrevious } from 'web.concurrency';
+import { debounce } from "@web/core/utils/timing";
+
+
+publicWidget.registry.AddressForm = publicWidget.Widget.extend({
+    selector: '.oe_cart .checkout_autoformat:has(input[name="street"][data-autocomplete-enabled="1"])',
+    events: {
+        'input input[name="street"]': '_onChangeStreet',
+        'click .js_autocomplete_result': '_onClickAutocompleteResult'
+    },
+
+    init: function() {
+        this.streetAndNumberInput = document.querySelector('input[name="street"]');
+        this.cityInput = document.querySelector('input[name="city"]');
+        this.zipInput = document.querySelector('input[name="zip"]');
+        this.countrySelect = document.querySelector('select[name="country_id"]');
+        this.stateSelect = document.querySelector('select[name="state_id"]');
+        this.dp = new DropPrevious();
+
+        this._onChangeStreet = debounce(this._onChangeStreet, 200);
+        this._super.apply(this, arguments);
+    },
+
+    _onChangeStreet: function(ev) {
+        this.dp.add(
+            this._rpc({
+                route: '/autocomplete/address',
+                params: {
+                    partial_address: ev.currentTarget.value
+                }
+            })).then((response) => {
+            if(response && response.template && response.template.trim()){
+                const inputContainer = ev.currentTarget.parentNode;
+                const dropdown = inputContainer.querySelector('.dropdown-menu');
+                if(dropdown) dropdown.remove();
+                inputContainer.appendChild($(response.template)[0]);
```

Comment:
Finally here I set the HTML manually from the template sent by the server. I think it should be fine as the template doesn't send back any user input, but again, correct me if I'm wrong.

Reply:
Why not have the server return the data and render the template client-side?